metricflow/query/query_parser.py

```diff
@@ -754,6 +758,25 @@ def _parse_linkable_elements(
             entity_specs=tuple(entity_specs),
         )
 
+    def _verify_resolved_granularity_for_date_part(
+        self,
+        requested_dimension_structured_name: StructuredLinkableSpecName,
+        partial_time_dimension_spec: PartialTimeDimensionSpec,
+        metric_references: Sequence[MetricReference],
+    ) -> None:
```

Comment:
Almost forgot, can we add a docblock explaining why this is here? Something like:

"""
Enforce that any granularity value associated with a date part query is the minimum.

By default, we will always ensure that a date_part query request uses the minimum granularity.
However, there are some interfaces where the user must pass in a granularity, so we need a check to
ensure that the correct value was passed in.
"""

Reply:
> there are some interfaces where the user must pass in a granularity

Which interface(s) require this ?